content/git/concepts/reset/reset.md

```diff
@@ -0,0 +1,65 @@
+---
+Title: 'Reset'
+Description: 'In Git, reset is used to set the HEAD pointer back to a previous commit and optionally undo staged changes and the working tree.'
+Subjects:
+  - 'Bash/Shell'
+  - 'Developer Tools'
+Tags:
+  - 'Git'
+  - 'Version Control'
+CatalogContent:
+  - 'learn-the-command-line'
+  - 'learn-git'
+---
+
+In Git, the `reset` command is used to change the state of the Git repository or undo commits.
+
+## Syntax
+
+In the terminal, the "git reset" command can be used by typing the following:
+
+```pseudo
+git reset <mode-option> <commit-reference-number>
+```
+
+### Mode Options
+
+The `<mode-options>` refers to how far `reset` will go when rolling back changes to a previous commit, including:
+
+- Where the `HEAD` is pointing towards (usually done with just `git reset`).
+- Whether the staging area or Index, reflects the commit the `HEAD` is now pointing towards.
+- Whether the working tree is rolled back to reflect the changes reset in the `HEAD` and staging area.
+
+More specifically, these options include:
+
+1. --soft: This undoes the commit but still shows the file as staged and permits only new commits.
+2. --mixed: This undoes the commit and unstages the file and the file can be edited and committed normally.
+3. --hard: This removes the new commit including all changes and uncommitted files.
+
+The commit can either be the commit hash generated by Git, or in reference to the HEAD.
+
+## Example
+
+Suppose a file named **example.txt** is accidentally committed to a Git repository. To undo the commit, the command `git reset --mixed HEAD~1` is used in the terminal. `--mixed` is the default mode, so `git reset HEAD~1` will achieve the same result. This resets the HEAD and index to the previous commit.
+
+**Note:** HEAD refers to the current commit while index refers to the staging area where files are [added using Git](https://www.codecademy.com/resources/docs/git/add).
+
+## After Committing
+
+```
+On branch main
+Your branch is up to date with 'origin/main'.
+nothing to commit, working tree clean
+```
+
+## After Resetting
```

Comment:
```suggestion
The text above indicates the following:

- The branch is up to date, with the `HEAD` pointing the the most recent commit.
- There is nothing to commit in the staging area.
- New changes haven't been made yet in the working tree.

To reset the `HEAD` pointer and the staging area back by 1 commit, one of the following commands can be run:
```

And then a code snippet, perhaps, with the two commands you'd mentioned in the paragraph above:

```git
git reset --mixed HEAD~1

git reset HEAD~1
```